qiskit/transpiler/passes/layout/sat_layout_for_commuting_gate_router.py

```diff
@@ -0,0 +1,301 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=too-many-function-args
+
+"""SATLayout pass to find a layout for commuting gates using subgraph isomorphism and a SAT solver"""
```

Comment:
Adding an example to the docs (either here or in the class) on how to use this pass would be great. What other passes does one need? How do you build, e.g., a QAOA pass manager with a SAT mapping.